WooCommerce: Update stock management UI

justinshreve opened the pull request:
This PR updates the stock management UI as described in #15165.

It removes the stock toggle and just always displays the stock input (if you a value is entered, you are managing stock).

It also moves the stock column to the first column in the variations table.

Fixes  #15165.

cc @kellychoffman 

To test:
* Go to `http://calypso.localhost:3000/store/product/:site`
* Verify that the stock quantity input always shows on both the variations editor and the simple card.
* Type a value in the simple card stock input and see backorder appear (since you are now managing stock).
* You can verify that `manage_stock` correctly gets set for variations using Redux dev tools. For the simple card, backorders displaying proves this.

belcherj commented on client/extensions/woocommerce/app/products/product-form-simple-card.js:
```diff
@@ -34,12 +33,10 @@ const ProductFormSimpleCard = ( { siteId, product, editProduct, translate } ) =>
 	const setStockQuantity = ( e ) => {
-		editProduct( siteId, product, { stock_quantity: e.target.value } );
+		const stock_quantity = Number( e.target.value ) >= 0 ? e.target.value : '';
+		const manage_stock = ( stock_quantity !== '' );
```
are these parens needed?

belcherj commented on client/extensions/woocommerce/app/products/product-form-variations-row.js:
```diff
@@ -64,7 +64,8 @@ class ProductFormVariationsRow extends Component {
 		const { siteId, editProductVariation, product, variation } = this.props;
 		const stock_quantity = Number( e.target.value ) >= 0 ? e.target.value : '';
-		editProductVariation( siteId, product, variation, { stock_quantity } );
+		const manage_stock = ( stock_quantity !== '' );
```
same here

justinshreve commented:
@kellychoffman 

>  I added placeholders 

Hah, I went back and forth on this before deploying -- guess I should have asked :).

> do you know why translate wouldn't be defined

I changed a line in my previous commit so translate just needs added back to that list of variables:

https://github.com/Automattic/wp-calypso/pull/15198/files#diff-475180105f8fc641fc846aedbf5c70d4L172

I can fix it while I look at jb's comment.
